### PR TITLE
Add dropwizard metrics to track status of pods

### PR DESCRIPTION
**Pull request by angry-cellophane**
This PR is to add some basic dropwizard metrics to track status of provisioned pods.

I migrated Jenkins instances from Mesos to Kubernetes recently and noticed that the kube plugin doesn't provide metrics related to builder nodes provisioning like the [mesos plugin does](https://github.com/jenkinsci/mesos-plugin/blob/mesos-0.18.1/src/main/java/org/jenkinsci/plugins/mesos/MesosCloud.java#L412).
These metrics become useful at scale when the number of Jenkins instance is quite big and they help to spot issues with Jenkins -> cloud provider as early as possible.
I took the same approach as the mesos plugin and use [dropwizard metrics](https://metrics.dropwizard.io/4.1.2/) (and [the metrics plugin](https://plugins.jenkins.io/metrics/)), however, I personally not a big fan of them as they don't have labels like [the prometheus metrics](https://plugins.jenkins.io/prometheus/) client. [Micrometer](https://micrometer.io) can be an alternative, it's like slf4j for logs. Any help with the metrics and what's the current stance in the jenkins world on them would be great.

### CODE REVIEW

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesCloud.java [hunk 1]
```diff
@@ -547,12 +550,14 @@ public KubernetesClient connect() throws KubernetesAuthException, IOException {
                 LOGGER.log(Level.FINEST, "Planned Kubernetes agents for template \"{0}\": {1}",
                         new Object[] { t.getName(), r.size() });
                 if (r.size() > 0) {
+                    Metrics.metricRegistry().counter("k8s.cloud.provision.nodes").inc(r.size());
```
Comment (Vlatombe) [inline src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesCloud.java:553]: Should be using `requestMetricNameFor` method on every call.

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesCloud.java [hunk 2]
```diff
@@ -1009,4 +1016,9 @@ public static void hpiRunInit() {
             }
         }
     }
+
+    private static String requestMetricNameFor(Label label) {
+        String labelText = (label == null) ? "nolabel" : label.getDisplayName();
+        return String.format("k8s.cloud.%s.provision.request", labelText);
```
Comment (Vlatombe) [inline src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesCloud.java:1022]:
```suggestion
        return String.format("kubernetes.cloud.%s.provision.request", labelText);
```

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesLauncher.java [hunk 3]
```diff
@@ -132,6 +133,7 @@ public synchronized void launch(SlaveComputer computer, TaskListener listener) {
             try {
                 pod = client.pods().inNamespace(namespace).create(pod);
             } catch (KubernetesClientException e) {
+                Metrics.metricRegistry().counter("k8s.cloud.pods.creation.failed").inc();
```
Comment (Vlatombe) [inline src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesLauncher.java:136]: Should be abstracting the key creation to a method such as `requestMetricNameFor`. No label here, but needs to be abstracted anyway.

**Review by Vlatombe — CHANGES_REQUESTED** [review]
These look like a good addition. I believe this needs to be slightly reworked in order to avoid too much repeats.

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesCloud.java (hunk 2 above)
Comment (angry-cellophane) [reply src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesCloud.java]: done

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesLauncher.java (hunk 3 above)
Comment (angry-cellophane) [reply src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesLauncher.java]: Thanks for looking into this PR and your suggestions.
Many metric names are static, so I moved then to a separate class and made most of them constants. There are a few that depend on some params, I put them in methods as you suggested.
Let me know if it looks better now and if you have any concern.

**Review by Vlatombe — APPROVED** [review]

**Pull request merged**